x-element.js

```diff
@@ -662,10 +662,48 @@ export default class XElement extends HTMLElement {
         observeMap.set(property, { value: undefined });
       }
     }
-    XElement.#hosts.set(host, {
+    const hostInfo = {
       initialized: false, reflecting: false, invalidProperties, listenerMap,
       renderRoot, render, template, properties, internal, computeMap,
       observeMap, defaultMap, valueMap,
+    };
+    XElement.#hosts.set(host, hostInfo);
+    XElement.#upgradeOwnProperties(host);
+    // Define properties during construction for React 19 introspection (#331).
```

Comment:
This is the crux of the change. We need to define properties during _construction_ so React (and other frameworks) can reliably detect that they should set values as properties (not attributes).